### PR TITLE
Fix integration tests + Restore them to the pipeline

### PR DESCRIPTION
**Pull request by jsimonsmartsheet**
This PR
1. Makes the integration tests compilable.
2. Adds compilation of the integration tests to the build
3. Marks off failing tests as ignore + fixes some tests
4. Removes unused broken docs generation project.

### CODE REVIEW

**Review by jsimonsmartsheet — COMMENTED** [review]
just testing authentication

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -33,8 +33,7 @@ jobs:
     - name: Run Mock API Tests
       working-directory: ./mock-api-test-sdk-net80
       run: |
-       dotnet build --configuration Release --no-restore
-       dotnet test --no-restore
```
Comment (jsimonsmartsheet) [inline .github/workflows/publish.yml:37]: already built

File: integration-test-sdk-net80/CellResourcesTest.cs [hunk 2]
```diff
@@ -13,14 +13,14 @@ public void TestCellResources()
             
             long sheetId = CreateSheet(smartsheet);
 
-            PaginatedResult<Column> columnsResult = smartsheet.SheetResources.ColumnResources.ListColumns(sheetId);
+            PaginatedResult<Column> columnsResult = smartsheet.SheetResources.ColumnResources.ListColumns(sheetId, null, null);
```
Comment (timothodge) [inline integration-test-sdk-net80/CellResourcesTest.cs]: I recently went through and tried to make our interfaces have optional parameters that didn't need to be set. Are these adding nulls showing that there are places where we need to have optionals also? 

Or, are these signs that my changes weren't enough?

File: integration-test-sdk-net80/SearchingResourcesTest.cs [hunk 3]
```diff
@@ -18,6 +19,8 @@ public void TestSearchingResources()
...
             // this very rarely shows up in 5 seconds
             System.Threading.Thread.Sleep(5000);
+
+            // FIXME: this 500's out "{\n  \"errorCode\" : 4000,\n  \"message\" : \"An unexpected error has occurred. Please contact the Support team at https://help.smartsheet.com/contact for assistance.\",\n  \"refId\" : \"sn4akh\",\n  \"internalStackTrace\" : \"com.navigo.system.core.CriticalException: error while calling the Restmethod:com.navigo.smartsheet.rest.service.v2.resources.SearchResource#topLevelSearch\\nRetry count: 0\\n\\tat com.navigo.smartsheet.rest.framework.v2.MethodInvoker.rethrowIfNotRetriable(MethodInvoker.java:352)\\n\\tat com.navigo.smartsheet.rest.framework.v2.MethodInvoker.invokeWithRetry(MethodInvoker.java:278)\\n\\tat com.navigo.smartsheet.rest.framework.v2.MethodInvoker.lambda$executeRequest$1(MethodInvoker.java:196)\\n\\tat com.navigo.smartsheet.rest.framework.v2.filter.ApiFilterChain.executeNext(ApiFilterChain.java:26)\\n\\tat com.navigo.smartsheet.rest.framework.v2.config.smartsheet.BIMetadataFilter.execute(BIMetadataFilter.java:34)\\n\\tat com.navigo.smartsheet.rest.framework.v2.filter.ApiFilterChain.executeNext...
```
Comment (timothodge) [inline integration-test-sdk-net80/SearchingResourcesTest.cs]: What do you mean this 500's out? Like we are leaking a stack trace?

File: smartsheet-csharp-sdk/main/Smartsheet/Api/SmartsheetBuilder.cs [hunk 4]
```diff
@@ -93,7 +93,8 @@ public class SmartsheetBuilder
         /// 
         /// <para>It is a constant with Value "https://api.smartsheet.com/2.0".</para>
         /// </summary>
-        public const string DEFAULT_BASE_URI = "https://api.smartsheet.com/2.0/";
```
Comment (timothodge) [inline smartsheet-csharp-sdk/main/Smartsheet/Api/SmartsheetBuilder.cs]: These changes need to be reverted correct?

File: integration-test-sdk-net80/CellResourcesTest.cs (hunk 2 above)
Comment (jsimonsmartsheet) [reply integration-test-sdk-net80/CellResourcesTest.cs]: I removed the two nulls 
`PaginatedResult<Column> columnsResult = smartsheet.SheetResources.ColumnResources.ListColumns(sheetId);`

and I get this from the compiler
```
The call is ambiguous between the following methods or properties: 
'SheetColumnResources.ListColumns(long, IEnumerable<ColumnInclusion>?, PaginationParameters?)' 
and 
'SheetColumnResources.ListColumns(long, IEnumerable<ColumnInclusion>?, PaginationParameters?, int?)'
```

so it looks like overridden functions don't play well with optional types.

File: smartsheet-csharp-sdk/main/Smartsheet/Api/SmartsheetBuilder.cs (hunk 4 above)
Comment (jsimonsmartsheet) [reply smartsheet-csharp-sdk/main/Smartsheet/Api/SmartsheetBuilder.cs]: whoops yeah, reverted. Is there a better way to set this via local environment variables?

File: integration-test-sdk-net80/SearchingResourcesTest.cs (hunk 3 above)
Comment (jsimonsmartsheet) [reply integration-test-sdk-net80/SearchingResourcesTest.cs]: Search was failing when I was testing in step box, but I pointed this to a prod account with a new API key and it works. I'm removing the `[Ignore]` tag.

File: integration-test-sdk-net80/CellResourcesTest.cs (hunk 2 above)
Comment (timothodge) [reply integration-test-sdk-net80/CellResourcesTest.cs]: So I believe we should get rid of the one without the int at the end....

That would remove ambiguousness. What do you think?

File: integration-test-sdk-net80/SearchingResourcesTest.cs (hunk 3 above)
Comment (timothodge) [reply integration-test-sdk-net80/SearchingResourcesTest.cs]: Sounds good!

File: smartsheet-csharp-sdk/main/Smartsheet/Api/SmartsheetBuilder.cs (hunk 4 above)
Comment (timothodge) [reply smartsheet-csharp-sdk/main/Smartsheet/Api/SmartsheetBuilder.cs]: Not that I know of :(

File: integration-test-sdk-net80/CellResourcesTest.cs (hunk 2 above)
Comment (jsimonsmartsheet) [reply integration-test-sdk-net80/CellResourcesTest.cs]: Yeah that makes sense. 

Anyone using the version without the optional int will get the default anyway - so we're good.

---

[that last parameter is 'level'](https://github.com/smartsheet/smartsheet-csharp-sdk/blob/mainline/smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/SheetColumnResourcesImpl.cs#L75)

https://smartsheet.redoc.ly/tag/columns
```
level
**Default: 1**

specifies whether multi-contact data is returned in a backwards-compatible, 
text format (**level=0**, default) or as multi-contact data (level=1).
```

Unrelated, but the documentation contradicts itself. 🤮

Comment (timothodge) [reply integration-test-sdk-net80/CellResourcesTest.cs]: Gross we should just add a ticket to tell the specific team to update that... 🤢

Comment (timothodge) [reply integration-test-sdk-net80/CellResourcesTest.cs]: Given that it now is one method with optionals do we feel like we want to get rid of the nulls?

**Review by timothodge — APPROVED** [review]
Happy with the changes, and thank you John for deep diving on optional parameters and other issues that were brought up.

**Pull request merged**